x/mongo/driver/batch_cursor.go

```diff
@@ -189,7 +189,7 @@ func NewBatchCursor(cr CursorResponse, clientSession *session.Client, clock *ses
 	if ds != nil {
 		bc.numReturned = int32(ds.DocumentCount())
 	}
-	if cr.Desc.WireVersion == nil || cr.Desc.WireVersion.Max < 4 {
```

Comment:
Shall we remove the entire `if` block from this line till L206?

Reply:
I think this should stay, it may still be possible for the wire version to not exist on the description. In this case, we would want the existing logic to be considered.